src/Bootstrap/RectorConfigsResolver.php

```diff
@@ -6,78 +6,88 @@
 
 use Rector\Core\ValueObject\Bootstrap\BootstrapConfigs;
 use Symfony\Component\Console\Input\ArgvInput;
-use Symplify\SetConfigResolver\ConfigResolver;
+use Symfony\Component\Console\Input\InputInterface;
+use Symplify\SmartFileSystem\Exception\FileNotFoundException;
 use Symplify\SmartFileSystem\SmartFileInfo;
 
 final class RectorConfigsResolver
 {
-    private ConfigResolver $configResolver;
+    public function provide(): BootstrapConfigs
+    {
+        $argvInput = new ArgvInput();
+        $mainConfigFileInfo = $this->resolveFromInputWithFallback($argvInput, 'rector.php');
 
-    /**
-     * @var array<string, SmartFileInfo[]>
-     */
-    private array $resolvedConfigFileInfos = [];
+        $rectorRecipeConfigFileInfo = $this->resolveRectorRecipeConfig($argvInput);
 
-    private SetConfigResolver $setConfigResolver;
+        $configFileInfos = [];
+        if ($rectorRecipeConfigFileInfo !== null) {
+            $configFileInfos[] = $rectorRecipeConfigFileInfo;
+        }
 
-    public function __construct()
-    {
-        $this->setConfigResolver = new SetConfigResolver();
-        $this->configResolver = new ConfigResolver();
+        return new BootstrapConfigs($mainConfigFileInfo, $configFileInfos);
     }
 
-    /**
-     * @return SmartFileInfo[]
-     */
-    public function resolveFromConfigFileInfo(SmartFileInfo $configFileInfo): array
+    private function resolveRectorRecipeConfig(ArgvInput $argvInput): ?SmartFileInfo
     {
-        $hash = sha1_file($configFileInfo->getRealPath());
-        if ($hash === false) {
-            return [];
+        if ($argvInput->getFirstArgument() !== 'generate') {
+            return null;
         }
 
-        if (isset($this->resolvedConfigFileInfos[$hash])) {
-            return $this->resolvedConfigFileInfos[$hash];
+        // autoload rector recipe file if present, just for \Rector\RectorGenerator\Command\GenerateCommand
+        $rectorRecipeFilePath = getcwd() . '/rector-recipe.php';
+        if (! file_exists($rectorRecipeFilePath)) {
+            return null;
         }
 
-        $setFileInfos = $this->setConfigResolver->resolve($configFileInfo);
-        $configFileInfos = array_merge([$configFileInfo], $setFileInfos);
+        return new SmartFileInfo($rectorRecipeFilePath);
+    }
+
+    private function resolveFromInput(ArgvInput $argvInput): ?SmartFileInfo
+    {
+        $configValue = $this->getOptionValue($argvInput, ['config', 'c']);
```

Comment:
There was a bug, there options have to be run with `--` and `-` too :face_with_head_bandage: 

Fix should be here:
https://github.com/rectorphp/rector-src/commit/037cfd435d69ea43dff9e44c17e252fdf40fa09a